docs/src/guide/periodic_problems.jl

```diff
@@ -0,0 +1,321 @@
+#     is actually arbitrary.
+
+
+
+# ## Adding potentials
+# So far so good. But free electrons are actually a little boring,
+# so let's add a potential interacting with the electrons.
+#
+# - The modified problem we will look at consists of diagonalising
+#   ```math
+#   H_k = \frac12 (-i \nabla + k)^2 + V
+#   ```
+#   for all ``k \in B`` with a periodic potential ``V`` interacting with the electrons.
+#
+# - A number of "standard" potentials are readily implemented in DFTK and
+#   can be assembled using the `terms` kwarg of the model.
+#   This allows to seamlessly construct
+#
+#   * density-functial theory models for treating electronic structures
+#     (see the [Tutorial](@ref)).
+#   * Gross-Pitaevskii models for bosonic systems
+#     (see [Gross-Pitaevskii equation in one dimension](@ref))
+#   * even some more unusual cases like anyonic models.
+#
+# In this tutorial we will go a little more low-level and directly provide
+# an analytic potential describing the interaction with the electrons to DFTK.
+#
+# First we define a custom Gaussian potential as a new "element" inside DFTK:
+
+struct ElementGaussian <: DFTK.Element
+    α  # Prefactor
+    L  # Extend
+end
+
+## Some default values
+ElementGaussian() = ElementGaussian(0.3, 10.0)
+
+## Real-space representation of a Gaussian
+function DFTK.local_potential_real(el::ElementGaussian, r::Real)
+    -el.α / (√(2π) * el.L) * exp(- (r / el.L)^2 / 2)
+end
+
+## Fourier-space representation of the Gaussian
+function DFTK.local_potential_fourier(el::ElementGaussian, q::Real)
+    ## = ∫ -α exp(-(r/L)^2 exp(-ir⋅q) dr
+    -el.α * exp(- (q * el.L)^2 / 2)
+end
+
+# A single potential looks like:
+
+using Plots
+using LinearAlgebra
+nucleus = ElementGaussian()
+plot(r -> DFTK.local_potential_real(nucleus, norm(r)), xlims=(-50, 50))
+
+# With this element at hand we can easily construct a setting
+# where two potentials of this form are located at positions
+# ``20`` and ``80`` inside the lattice ``[0, 100]``:
+
+using LinearAlgebra
+
+## Define the 1D lattice [0, 100]
+lattice = diagm([100., 0, 0])
+
+## Place them at 20 and 80 in *fractional coordinates*,
+## that is 0.2 and 0.8, since the lattice is 100 wide.
+nucleus = ElementGaussian()
+atoms = [nucleus => [[0.2, 0, 0], [0.8, 0, 0]]]
+
+## Assemble the model, discretise and build the Hamiltonian
+model = Model(lattice; atoms=atoms, terms=[Kinetic(), AtomicLocal()])
+basis = PlaneWaveBasis(model; Ecut=300, kgrid=(1, 1, 1));
+ham   = Hamiltonian(basis)
+
+## Extract the total potential term of the Hamiltonian and plot it
+potential = DFTK.total_local_potential(ham)[:, 1, 1]
+rvecs = collect(r_vectors_cart(basis))[:, 1, 1]  # slice along the x axis
+x = [r[1] for r in rvecs]                        # only keep the x coordinate
+plot(x, potential, label="", xlabel="x", ylabel="V(x)")
+
+# Notice how DFTK took care of the periodic wrapping of the potential values going
+# from ``0`` and ``100``.
+#
```

Comment:
I would maybe make the point that the potential is the sum of local ("atomic") potentials, and that the atoms are implicitly periodized